tests/languages/golang_test.py

```diff
@@ -165,3 +166,72 @@ def test_during_commit_all(tmp_path, tempdir_factory, store, in_git_dir):
         fn=cmd_output_mocked_pre_commit_home,
         tempdir_factory=tempdir_factory,
     )
+
+
+def test_automatic_toolchain_switching(tmp_path):
+    go_mod = '''\
+module toolchain-version-test
+
+go 1.23.1
+'''
+    main_go = '''\
+package main
+
+func main() {}
+'''
+    tmp_path.joinpath('go.mod').write_text(go_mod)
+    mod_dir = tmp_path.joinpath('toolchain-version-test')
+    mod_dir.mkdir()
+    main_file = mod_dir.joinpath('main.go')
+    main_file.write_text(main_go)
+
+    with pytest.raises(CalledProcessError) as excinfo:
+        run_language(
+            path=tmp_path,
+            language=golang,
+            version='1.22.0',
+            exe='golang-version-test',
+        )
+
+        assert 'go.mod requires go >= 1.23.1' in excinfo.value
+
+
+def test_automatic_toolchain_switching_go_fmt(tmp_path, monkeypatch):
+    go_mod_hook = '''\
+module toolchain-version-test
+
+go 1.22.0
+'''
+    go_mod = '''\
+module toolchain-version-test
+
+go 1.23.1
+'''
+    main_go = '''\
+package main
+
+func main() {}
+'''
+    hook_dir = tmp_path.joinpath('hook')
+    hook_dir.mkdir()
+    hook_dir.joinpath('go.mod').write_text(go_mod_hook)
+
+    test_dir = tmp_path.joinpath('test')
+    test_dir.mkdir()
+    test_dir.joinpath('go.mod').write_text(go_mod)
+    main_file = test_dir.joinpath('main.go')
+    main_file.write_text(main_go)
+
+    with monkeypatch.context() as m:
+        m.chdir(str(test_dir))
+
+        ret, out = run_language(
+            path=hook_dir,
+            language=golang,
+            version='1.22.0',
+            exe='go fmt',
+            file_args=(str(main_file)),
```

Comment:
this is just a parenthesized string and should be a tuple I believe -- I'm kinda surprised the type checker was ok with this? (oh duh, `str` is a `Sequence[str]` so it just kinda lets it happen)